src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -123,14 +115,13 @@ private void checkIndex(int index, int maxAvailableIndex) {
     }
 
     private void addLast(T value) {
-        Node<T> prevNode = tail;
-        Node<T> newNode = new Node<>(prevNode, value, null);
-        if (prevNode == null) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
             head = newNode;
         } else {
-            prevNode.next = newNode;
+            tail.next = newNode;
         }
-        tail = newNode;
+        this.tail = newNode;
```

Comment:
why `this`?

Reply:
My bad((( It was IDE autocorrection which I didnt see